Implementing selecting every event per view in calendar

Jrende opened the pull request:
## Description
This PR add buttons for selecting all events when selecting events in calendar.
The select all button will select all events on the current view, so on the month it will select all for the focused month. The deselect all button will deselect all selected events.

## Screenshots
https://github.com/user-attachments/assets/9d8733b0-7b3e-42ed-8082-81d1ca8891cc
Switching between months, selecting events per month.

https://github.com/user-attachments/assets/556c3b14-bb10-4d13-8dca-52d527830842
Use together with filters

## Changes
* Adds select all  button for selecting all currently visible events
* Add a list for the event store containing all visible events, which is updated by individual calendar views.


## Notes to reviewer


## Related issues
Resolves #2755

richardolsson reviewed (commented):
This is such great work. I really like the UX of this feature, but I do feel a little bit uncomfortable with the implementation, especially the duplication of state.

I see that the state for filters is already in the store, so I'm wondering if it would be possible to calculate what's visible from the state already available in store. Perhaps it's missing the state of the calendar view, i.e. which view mode and what focus date?

What do you think?

Jrende commented:
I rewrote the functionality without duplicating state, which turned out to be much more code than before. I created a store for calendar to save focus date, timescale and other data needed to fetch the events.

I'm not sure if I'm using stores in the most clean way, and I had some trouble handling hooks.
When using a store, i first need to update it with values for focusDate etc. Ideally react would wait until that is done before fetching, but since it fetches the events using a hook it's impossible to put that in an async function. Maybe there's some magic using redux but I couldn't figure it out. Instead I modified the functions to return a empty response if needed, for example in `useMonthCalendarEvents`.

This is a lot more code than my earlier implementation, and I'd be happy to know if there are better ways of doing this.

ziggabyte commented:
This has been open for a really long time now, I think the reason is that it's not been obvious how to give the best feedback on this to make it work and give it the best way forward. So, I'm going to close this PR and let it exist as documentation of a way that was attempted, things that were learned - and then I'm going to expand the text in the issue to include some things that have been learned by your work here, and what can be ways to move forward. 

Very good job on exploring this and trying different things, that made us all smarter and learn how this can be solved in a good way! And thank you for being patient in waiting for a review!
